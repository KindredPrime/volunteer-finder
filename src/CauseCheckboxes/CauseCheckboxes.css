.CauseCheckboxes__cause-wrapper {
  position: relative;
  margin-bottom: 5px;
}

.CauseCheckboxes label {
  margin-left: auto;
  margin-right: auto;
  cursor: pointer;
}

.CauseCheckboxes input {
  position: relative;
  left: -5px;
  opacity: 0;
  cursor: pointer;
}

.CauseCheckboxes__custom-checkbox {
  display: inline-block;
  position: relative;
  left: 12px;
  top: 4px;
  width: 13px;
  height: 13px;
  background-color: white;
  border-radius: 2px;
  opacity: 0.75;
}

.CauseCheckboxes__custom-checkbox.checked {
  background-color: var(--accent-color);
  opacity: 1;
}

.CauseCheckboxes__checkmark {
  position: relative;
  color: black;
  top: -5px;
  height: 15px;
  font-size: 12px;
  opacity: 0;
}

.CauseCheckboxes__checkmark.checked {
  opacity: 1;
}