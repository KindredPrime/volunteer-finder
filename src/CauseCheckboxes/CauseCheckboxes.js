import React from 'react';
import PropTypes from 'prop-types';

/*
  Used to render checkboxes for causes
*/
function CauseCheckboxes(props) {
  const { causes, handleClick, legend } = props;

  return (
    <fieldset className="CauseCheckboxes">
      <legend>{legend}</legend>

      {causes.map((cause) => (
        <div key={cause.id}>
          <label htmlFor={`causes-${cause.id}`}>{cause.cause_name}</label>
          <input 
            type="checkbox"
            id={`causes-${cause.id}`}
            name="causes"
            value={cause.cause_name}
            onChange={() => handleClick(cause.cause_name)}
            checked={cause.checked}
          />
        </div>
      ))}
    </fieldset>
  );
}

CauseCheckboxes.propTypes = {
  causes: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    cause_name: PropTypes.string
  })).isRequired,
  handleClick: PropTypes.func.isRequired,
  legend: PropTypes.string.isRequired
}

export default CauseCheckboxes;