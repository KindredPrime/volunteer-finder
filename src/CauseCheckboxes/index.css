.CauseCheckboxes__cause-wrapper {
  position: relative;
  margin-bottom: 5px;
}

.CauseCheckboxes label {
  margin-left: auto;
  margin-right: auto;
  cursor: pointer;
}

.CauseCheckboxes input {
  position: relative;
  left: -5px;

  /* Hides the default checkbox input */
  opacity: 0;
  cursor: pointer;
}

/*
  Overlays a custom checkbox input on top of the default one
*/
.CauseCheckboxes__custom-checkbox {
  display: inline-block;
  position: relative;
  left: 12px;
  top: 4px;
  width: 13px;
  height: 13px;
  background-color: white;
  border-radius: 2px;
  opacity: 0.75;
}

.CauseCheckboxes__custom-checkbox.checked {
  background-color: var(--accent-color);
  opacity: 1;
}

/*
  Custom checkmark, hidden by default
*/
.CauseCheckboxes__checkmark {
  position: relative;
  color: black;
  top: -5px;
  height: 15px;
  font-size: 12px;
  opacity: 0;
}

/*
  When the custom checkbox is checked, show the custom checkmark
*/
.CauseCheckboxes__custom-checkbox.checked .CauseCheckboxes__checkmark {
  opacity: 1;
}