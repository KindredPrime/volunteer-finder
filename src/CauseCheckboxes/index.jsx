import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCheck } from '@fortawesome/free-solid-svg-icons';
import './index.css';

/*
  Render checkbox inputs for causes
*/
function CauseCheckboxes(props) {
  // checkedCauses is an object where the keys are the names of causes that have been checked, and
  // the values are all set to 'true'
  const { causes, checkedCauses, handleClick, legend } = props;

  const customCheckboxClass = 'CauseCheckboxes__custom-checkbox';

  // Add the checked status to the list of all causes
  const causesWithChecks = causes.map((cause) => {
    return checkedCauses[cause.cause_name]
      ? {
        ...cause,
        checked: true
      }
      : {
        ...cause,
        checked: false
      };
  })

  return (
    <fieldset className="CauseCheckboxes">
      <legend>{legend}</legend>

      {causesWithChecks.map((cause) => (
        <div className="CauseCheckboxes__cause-wrapper" key={cause.id}>
          <label htmlFor={`causes-${cause.id}`}>
            {cause.cause_name}
          </label>
          <div
            className={!cause.checked
              ? customCheckboxClass
              : `${customCheckboxClass} checked`}
          >
            <FontAwesomeIcon
              className="CauseCheckboxes__checkmark"
              icon={faCheck}
            />
          </div>
          <input
            type="checkbox"
            id={`causes-${cause.id}`}
            name="causes"
            value={cause.cause_name}
            onChange={() => handleClick(cause.cause_name)}
            checked={cause.checked}
          />
        </div>
      ))}
    </fieldset>
  );
}

CauseCheckboxes.propTypes = {
  causes: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    cause_name: PropTypes.string
  })).isRequired,
  handleClick: PropTypes.func.isRequired,
  legend: PropTypes.string.isRequired
}

export default CauseCheckboxes;