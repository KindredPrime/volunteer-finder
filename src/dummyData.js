const dummyUsers = [
  {
    id: 1,
    email: 'email1@email.com',
    username: 'user1',
    password: 'password1',
    orgsAdded: [1],
    eventsAdded: [1, 2]
  },
  {
    id: 2,
    email: 'email2@email.com',
    username: 'user2',
    password: 'password2',
    orgsAdded: [2],
    eventsAdded: [3]
  },
  {
    id: 3,
    email: 'email3@email.com',
    username: 'user3',
    password: 'password3',
    orgsAdded: [3],
    eventsAdded: []
  },
  {
    id: 4,
    email: 'email4@email.com',
    username: 'user4',
    password: 'password4',
    orgsAdded: [],
    eventsAdded: []
  },
  {
    id: 5,
    email: 'email5@email.com',
    username: 'user5',
    password: 'password5',
    orgsAdded: [4],
    eventsAdded: [4]
  }
];

const dummyOrgs = [
  {
    id: 1,
    name: 'YMCA',
    website: 'https://www.ymca.com',
    phone: '1-800-555-7777',
    email: 'contact@ymca.com',
    address: '1 YMCA Street Real City, Real State',
    causes: [1, 2, 3],
    tags: [2, 3, 4, 6]
  },
  {
    id: 2,
    name: 'ASPCA',
    website: 'https://www.aspca.com',
    phone: '1-800-123-4567',
    email: 'contact@aspca.com',
    address: '1 ASPCA Street Real City, Real State',
    causes: [5, 2],
    tags: [2, 3, 4, 10]
  },
  {
    id: 3,
    name: 'National VOAD',
    website: 'https://www.nvoad.com',
    phone: '1-800-456-7890',
    email: 'contact@nvoad.com',
    address: '1 NVOAD Street Rivendell, Middle Earth',
    causes: [4, 2],
    tags: [2, 3, 4, 5, 7, 11]
  },
  {
    id: 4,
    name: 'National Air and Space Museum',
    website: 'https://www.airandspace.com',
    phone: '1-800-987-6543',
    email: 'contact@airandspace.com',
    address: '1 Space Street Air City, Outer Space',
    causes: [6, 7],
    tags: [1, 2, 3, 4, 6, 12]
  }
];

const dummyEvents = [
  {
    id: 1,
    name: 'Member Appreciation Day',
    organization: 1,
    location: 'Reston YMCA Center',
    date: 'February 20, 2020 00:00:00',
    duration: '4 hours',
    description: 'Greet YMCA members as they enter the facility and offer them snacks.  See https://www.website.com for more details.',
    causes: [2, 3],
    tags: [2, 4, 12]
  },
  {
    id: 2,
    name: 'Triathlon Assistance',
    organization: 1,
    location: 'Reston YMCA Center',
    date: 'January 27, 2020 00:00:00',
    duration: '4 hours',
    description: 'Help YMCA staff host their annual indoor triathlon: sign in triathlon participants, direct the participants to the different stages of the triathlon, and record times. See https://www.triathlon.com for more details.',
    causes: [2, 3],
    tags: [2, 3, 9]
  },
  {
    id: 3,
    name: 'Adoption Day',
    organization: 2,
    location: 'Times Square, New York City, New York',
    date: 'December 13, 2020 00:00:00',
    duration: '8 hours',
    description: 'Help the ASPCA run their annual adoption day for their local pet shelter.',
    causes: [5],
    tags: [2, 3, 9, 12]
  },
  {
    id: 4,
    name: 'Docent Training Class',
    organization: 4,
    location: '14390 Air and Space Museum Pkwy Chantilly, VA 20151',
    date: 'January 30, 2021 00:00:00',
    duration: '2 hours',
    description: 'This is a required class for anyone who wants to be a docent (tour guide) at the National Air and Space Museum.',
    causes: [6, 7],
    tags: [2, 3, 13]
  }
];

const dummyTags = [
  {
    id: 1,
    name: 'virtual'
  },
  {
    id: 2,
    name: 'in-person'
  },
  {
    id: 3,
    name: 'weekends'
  },
  {
    id: 4,
    name: 'weekdays'
  },
  {
    id: 5,
    name: 'nights'
  },
  {
    id: 6,
    name: 'group'
  },
  {
    id: 7,
    name: 'delivery'
  },
  {
    id: 8,
    name: 'cooking'
  },
  {
    id: 9,
    name: 'administration'
  },
  {
    id: 10,
    name: 'cleaning'
  },
  {
    id: 11,
    name: 'heavy lifting'
  },
  {
    id: 12,
    name: 'greeting'
  },
  {
    id: 13,
    name: 'training'
  }
];

const dummyCauses = [
  {
    id: 1,
    name: 'Youth'
  },
  {
    id: 2,
    name: 'Community'
  },
  {
    id: 3,
    name: 'Health'
  },
  {
    id: 4,
    name: 'Disaster Response'
  },
  {
    id: 5,
    name: 'Animals'
  },
  {
    id: 6,
    name: 'Museums'
  },
  {
    id: 7,
    name: 'Education'
  },
  {
    id: 8,
    name: 'Human Rights'
  },
  {
    id: 9,
    name: 'Arts'
  }
];

export {
  dummyUsers,
  dummyOrgs,
  dummyEvents,
  dummyTags,
  dummyCauses
};