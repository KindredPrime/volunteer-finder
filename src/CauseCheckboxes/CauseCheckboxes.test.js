import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import CauseCheckboxes from './CauseCheckboxes';
import { dummyCauses } from '../dummyData';

describe('CauseCheckboxes Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <CauseCheckboxes 
        causes={[]}
        handleClick={() => {}}
        legend="Legend"
      />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    render(
      <CauseCheckboxes
        causes={dummyCauses}
        handleClick={() => {}}
        legend="Causes"
      />
    );

    expect(document.body).toMatchSnapshot();
  });
});