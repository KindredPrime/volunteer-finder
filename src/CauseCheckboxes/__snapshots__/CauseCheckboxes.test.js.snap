// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CauseCheckboxes Component renders the UI as expected 1`] = `
<body>
  <div>
    <fieldset
      class="CauseCheckboxes"
    >
      <legend>
        Causes
      </legend>
      <div>
        <label
          for="causes-1"
        >
          Youth
        </label>
        <input
          id="causes-1"
          name="causes"
          type="checkbox"
          value="Youth"
        />
      </div>
      <div>
        <label
          for="causes-2"
        >
          Health
        </label>
        <input
          id="causes-2"
          name="causes"
          type="checkbox"
          value="Health"
        />
      </div>
      <div>
        <label
          for="causes-3"
        >
          Disaster Response
        </label>
        <input
          id="causes-3"
          name="causes"
          type="checkbox"
          value="Disaster Response"
        />
      </div>
      <div>
        <label
          for="causes-4"
        >
          Animals
        </label>
        <input
          id="causes-4"
          name="causes"
          type="checkbox"
          value="Animals"
        />
      </div>
      <div>
        <label
          for="causes-5"
        >
          History
        </label>
        <input
          id="causes-5"
          name="causes"
          type="checkbox"
          value="History"
        />
      </div>
      <div>
        <label
          for="causes-6"
        >
          Education
        </label>
        <input
          id="causes-6"
          name="causes"
          type="checkbox"
          value="Education"
        />
      </div>
      <div>
        <label
          for="causes-7"
        >
          Human Rights
        </label>
        <input
          id="causes-7"
          name="causes"
          type="checkbox"
          value="Human Rights"
        />
      </div>
      <div>
        <label
          for="causes-8"
        >
          Arts
        </label>
        <input
          id="causes-8"
          name="causes"
          type="checkbox"
          value="Arts"
        />
      </div>
      <div>
        <label
          for="causes-9"
        >
          Wildlife and Environment
        </label>
        <input
          id="causes-9"
          name="causes"
          type="checkbox"
          value="Wildlife and Environment"
        />
      </div>
      <div>
        <label
          for="causes-10"
        >
          Elderly
        </label>
        <input
          id="causes-10"
          name="causes"
          type="checkbox"
          value="Elderly"
        />
      </div>
      <div>
        <label
          for="causes-11"
        >
          Poverty
        </label>
        <input
          id="causes-11"
          name="causes"
          type="checkbox"
          value="Poverty"
        />
      </div>
    </fieldset>
  </div>
</body>
`;
